packages/material-ui/src/InputBase/Textarea.js

```diff
@@ -45,12 +45,18 @@ const Textarea = React.forwardRef(function Textarea(props, ref) {
     inputShallow.value = input.value || props.placeholder || 'x';
 
     // The height of the inner content
-    const innerHeight = inputShallow.scrollHeight;
+    let innerHeight = inputShallow.scrollHeight;
     const boxSizing = computedStyle['box-sizing'];
 
     // Measure height of a textarea with a single row
-    inputShallow.value = 'x';
-    const singleRowHeight = inputShallow.scrollHeight;
+    inputShallow.value = 'y';
+    const singleRowHeight = getStyleValue(window.getComputedStyle(inputShallow), 'line-height');
+    const rowsCount = Math.floor(innerHeight / singleRowHeight);
+    const descendersHeight = inputShallow.scrollHeight - singleRowHeight;
+
+    // Force one-line multiline input to have the same height as a non-multiline input
+    // but add the descenders height otherwise
+    innerHeight = rowsCount <= 1 ? singleRowHeight : rowsCount * singleRowHeight + descendersHeight;
```

Comment:
How can we reproduce the problem solved by this logic?

Reply:
Remove this line => one-line input has height of 20px (instead of 19px) and >1 line input will change its height when adding a new line and then entering the first letter in that line.

Reply:
Usign the issue codesandbox, I can see the following:
![Capture d’écran 2019-05-26 à 16 31 59](https://user-images.githubusercontent.com/3165635/58383186-df23c080-7fd3-11e9-9bf2-799dd76816c3.png)

The multiple={true} and multiple={false} inputs have the same height: 19px. What do I miss?

Reply:
It has a height of 33px vs 32px for me.

Reply:
Interesting. I'm using Chrome 74 on MacOS.

Reply:
It's the other way around on my phone (smaller instead of largeur) 
![Screenshot_20190526-170534](https://user-images.githubusercontent.com/3165635/58383547-8efb2d00-7fd8-11e9-9971-c85dd794bc0a.jpg)

Reply:
Looks normal for me too (non-high-dpi display)

Reply:
How does the fixed version look for you? :wink: It fixed it for me.

Reply:
Chrome, Firefox and Safari aren't impacted on my computer, the height is 19px. However, if I open it with Chrome on my phone, the height is 18px. Should we care about an obscure 1px difference?

Reply:
Yes, we should. Because _that is the bug_: the multiline input is 1px higher (in some browsers) than the non-multiline input.

As I said, this PR fixes the issue for me. Can we just skip this discussion and verify if it fixes the problem?

Reply:
This pull request fixes the problem on my phone but introduces this regression:

**Before**
![Capture d’écran 2019-05-27 à 09 41 35](https://user-images.githubusercontent.com/3165635/58403855-00cc8880-8064-11e9-9458-5e1f7b136e2a.png)

**After**
![Capture d’écran 2019-05-27 à 09 41 47](https://user-images.githubusercontent.com/3165635/58403888-15a91c00-8064-11e9-9db1-1d1c0cfcacb6.png)

(The second textarea height is too high).

Could you try this change? It covers all the use cases in the environments I can test:

```diff
diff --git a/packages/material-ui/src/InputBase/Textarea.js b/packages/material-ui/src/InputBase/Textarea.js
index 8c72dfa8a..d018f476b 100644
--- a/packages/material-ui/src/InputBase/Textarea.js
+++ b/packages/material-ui/src/InputBase/Textarea.js
@@ -9,19 +9,6 @@ function getStyleValue(computedStyle, property) {

 const useEnhancedEffect = typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;

-const styles = {
-  /* Styles applied to the shadow textarea element. */
-  shadow: {
-    // Visibility needed to hide the extra text area on iPads
-    visibility: 'hidden',
-    // Remove from the content flow
-    position: 'absolute',
-    // Ignore the scrollbar width
-    overflow: 'hidden',
-    height: '0',
-  },
-};
-
 /**
  * @ignore - internal component.
  *
@@ -33,30 +20,23 @@ const Textarea = React.forwardRef(function Textarea(props, ref) {
   const { current: isControlled } = React.useRef(value != null);
   const inputRef = React.useRef(null);
   const [state, setState] = React.useState({});
-  const shadowRef = React.useRef(null);
   const handleRef = useForkRef(ref, inputRef);

   const syncHeight = React.useCallback(() => {
     const input = inputRef.current;
-    const inputShallow = shadowRef.current;
+    const savedHeight = input.style.height;
+    const savedOverflow = input.style.overflow;
+    input.style.overflow = 'hidden';
+    input.style.height = '0';

     const computedStyle = window.getComputedStyle(input);
-    inputShallow.style.width = computedStyle.width;
-    inputShallow.value = input.value || props.placeholder || 'x';

     // The height of the inner content
-    let innerHeight = inputShallow.scrollHeight;
+    const innerHeight = input.scrollHeight;
     const boxSizing = computedStyle['box-sizing'];

     // Measure height of a textarea with a single row
-    inputShallow.value = 'y';
-    const singleRowHeight = getStyleValue(window.getComputedStyle(inputShallow), 'line-height');
-    const rowsCount = Math.floor(innerHeight / singleRowHeight);
-    const descendersHeight = inputShallow.scrollHeight - singleRowHeight;
-
-    // Force one-line multiline input to have the same height as a non-multiline input
-    // but add the descenders height otherwise
-    innerHeight = rowsCount <= 1 ? singleRowHeight : rowsCount * singleRowHeight + descendersHeight;
+    const singleRowHeight = getStyleValue(window.getComputedStyle(input), 'line-height');

     // The height of the outer content
     let outerHeight = innerHeight;
@@ -79,6 +59,9 @@ const Textarea = React.forwardRef(function Textarea(props, ref) {
         getStyleValue(computedStyle, 'border-top-width');
     }

+    input.style.overflow = savedOverflow;
+    input.style.height = savedHeight;
+
     setState(prevState => {
       // Need a large enough different to update the height.
       // This prevents infinite rendering loop.
@@ -91,7 +74,7 @@ const Textarea = React.forwardRef(function Textarea(props, ref) {

       return prevState;
     });
-  }, [setState, rows, rowsMax, props.placeholder]);
+  }, [setState, rows, rowsMax]);

   React.useEffect(() => {
     const handleResize = debounce(() => {
@@ -120,27 +103,17 @@ const Textarea = React.forwardRef(function Textarea(props, ref) {
   };

   return (
-    <React.Fragment>
-      <textarea
-        value={value}
-        onChange={handleChange}
-        ref={handleRef}
-        style={{
-          height: state.outerHeight,
-          overflow: state.outerHeight === state.innerHeight ? 'hidden' : null,
-          ...style,
-        }}
-        {...other}
-      />
-      <textarea
-        aria-hidden
-        className={props.className}
-        readOnly
-        ref={shadowRef}
-        tabIndex={-1}
-        style={{ ...styles.shadow, ...style }}
-      />
-    </React.Fragment>
+    <textarea
+      value={value}
+      onChange={handleChange}
+      ref={handleRef}
+      style={{
+        height: state.outerHeight,
+        overflow: state.outerHeight === state.innerHeight ? 'hidden' : null,
+        ...style,
+      }}
+      {...other}
+    />
   );
 });
```

Also, can we focus on the padding issue here and have another pull request for the Textarea change? The CSS change is straightforward, it will provide 80% of the value, it's unlikely we will have to revert it. It's a different story for the Textarea component, it's very hard to get the logic work everywhere.

Reply:
> Otherwise, you could just close the original bug report as "won't fix" if we (as a front-end library) don't care about 1px differences.

In the original issue, how many pixels different to we have? 3px in the standard variant, 7px in the outlined variant, 7px in the filled case.

Reply:
@oliviertassinari I'll try your suggested fix and add a regression test. As I already said before, fixing the padding is not 80% of this bug in my opinion. Let's fix 100% of the bug. This is a UI library, why don't you care about them pixels? :smile:

Reply:
I don't believe in 100%, I think that we should focus on doing the enough thing: https://fs.blog/jason-fried/

Reply:
I don't care. I won't waste our time by discussing this further or create a new PR.

Reply:
Time is our most important resource. I agree that we should put it at good use.

Reply:
The regression was due to `box-sizing: content-box` on the text area. It's fixed now. :)

Reply:
Does the proposed fix solve the problem in your env?

Reply:
Yes, it does.